### PR TITLE
app: modules: POC split TRIGGER_CHAN into separate channels

### PR DESCRIPTION
**Pull request by gregersrygg**
Instead of using the TRIGGER_CHAN message for a type. Use separate channels for data sample and poll, and only notify observers instead of publishing a message on the channels.

### CODE REVIEW

File: app/src/common/message_channel.c [hunk 1]
```diff
@@ -18,6 +18,22 @@ ZBUS_CHAN_DEFINE(TRIGGER_CHAN,
 		 ZBUS_MSG_INIT(0)
 );
 
+ZBUS_CHAN_DEFINE(DATA_SAMPLE_CHAN,
+		 uint8_t,
```
Comment (balaji-nordic) [inline app/src/common/message_channel.c:22]: Remove trigger channel above

File: app/src/common/message_channel.h [hunk 2]
```diff
@@ -88,7 +88,9 @@ ZBUS_CHAN_DECLARE(
 	NETWORK_CHAN,
 	PAYLOAD_CHAN,
 	TIME_CHAN,
-	TRIGGER_CHAN
+	TRIGGER_CHAN,
```
Comment (balaji-nordic) [inline app/src/common/message_channel.h:91]: Remove

Comment (gregersrygg) [reply app/src/common/message_channel.h:91]: Other modules still use it. I just modified a couple of modules to show an alternative way of triggering without sending a message.

**Comment by simensrostad** [comment]
@gregersrygg Can this be closed?

**Pull request closed without merging**